components/builder-api-client/src/builder.rs

```diff
@@ -172,25 +172,24 @@ impl BuilderAPIClient {
         resp.ok_if(StatusCode::OK)?;
 
         fs::create_dir_all(&dst_path)?;
-
         let file_name = resp.get_header(X_FILENAME)?;
         let dst_file_path = dst_path.join(file_name);
         let w = AtomicWriter::new(&dst_file_path)?;
         w.with_writer(|mut f| {
              match progress {
                  Some(mut progress) => {
-                     let size = resp.get_header(CONTENT_LENGTH)?
-                                    .parse()
-                                    .map_err(Error::ParseIntError)?;
-
-                     progress.size(size);
+                     // There will be no CONTENT_LENGTH header if an on prem
+                     // builder is using chunked transfer encoding
+                     if let Ok(header) = resp.get_header(CONTENT_LENGTH) {
+                         let size = header.parse().map_err(Error::ParseIntError)?;
+                         progress.size(size);
+                     }
```

Comment:
I notice in both this case and the one above, when `progress` is `None`, we just call `io::copy`.  In this case, we're creating a `BroadcastWriter` that takes `progress` but if there's no `CONTENT_LENGTH` header, then `progress` never has its `size` set, right?

Phrased differently, in the case where there's no `CONTENT_LENGTH` length header, and thus we can't size `progress`, does it make sense to just call `io::copy` like we do when `progress` is `None`, and skip creating the `BroadcastWriter`?

Reply:
good point. I just changed this to bail on the `BroadcastWriter` when there is no content length.